tests/python/unittest/test_gluon_rnn.py

```diff
@@ -152,190 +124,7 @@ def test_lstmp():
-
-    args, outs, auxs = outputs.infer_shape(t0_data=(10, 50), t1_data=(10, 50))
-    assert outs == [(10, 50), (10, 50)]
-    outputs = outputs.eval(**{'t0_data': mx.nd.ones((10, 50)),
-                              't1_data': mx.nd.ones((10, 50)),
-                              cell.base_cell.i2h_weight.var().name: mx.nd.zeros((150, 50)),
-                              cell.base_cell.i2h_bias.var().name: mx.nd.zeros((150, )),
-                              cell.base_cell.h2h_weight.var().name: mx.nd.zeros((150, 50)),
-                              cell.base_cell.h2h_bias.var().name: mx.nd.zeros((150, ))})
-    expected_outputs = np.ones((10, 50))
-    assert np.array_equal(outputs[0].asnumpy(), expected_outputs)
-    assert np.array_equal(outputs[1].asnumpy(), expected_outputs)
-
-
-@pytest.mark.serial
-def test_residual_bidirectional():
-    cell = gluon.rnn.ResidualCell(
-            gluon.rnn.BidirectionalCell(
-                gluon.rnn.GRUCell(25),
-                gluon.rnn.GRUCell(25)))
-    inputs = [mx.sym.Variable('rnn_t%d_data'%i) for i in range(2)]
-    outputs, _ = cell.unroll(2, inputs, merge_outputs=False)
-    outputs = mx.sym.Group(outputs)
-    params = cell.collect_params()
-    assert sorted(params.keys()) == \
-           ['base_cell.l_cell.h2h_bias', 'base_cell.l_cell.h2h_weight',
-            'base_cell.l_cell.i2h_bias', 'base_cell.l_cell.i2h_weight',
-            'base_cell.r_cell.h2h_bias', 'base_cell.r_cell.h2h_weight',
-            'base_cell.r_cell.i2h_bias', 'base_cell.r_cell.i2h_weight']
-    # assert outputs.list_outputs() == \
-    #        ['bi_t0_plus_residual_output', 'bi_t1_plus_residual_output']
-
-    args, outs, auxs = outputs.infer_shape(rnn_t0_data=(10, 50), rnn_t1_data=(10, 50))
-    assert outs == [(10, 50), (10, 50)]
-    outputs = outputs.eval(**{'rnn_t0_data':mx.nd.ones((10, 50))+5,
-                              'rnn_t1_data':mx.nd.ones((10, 50))+5,
-                              cell.base_cell.l_cell.i2h_weight.var().name:mx.nd.zeros((75, 50)),
-                              cell.base_cell.l_cell.i2h_bias.var().name:mx.nd.zeros((75,)),
-                              cell.base_cell.l_cell.h2h_weight.var().name:mx.nd.zeros((75, 25)),
-                              cell.base_cell.l_cell.h2h_bias.var().name:mx.nd.zeros((75,)),
-                              cell.base_cell.r_cell.i2h_weight.var().name:mx.nd.zeros((75, 50)),
-                              cell.base_cell.r_cell.i2h_bias.var().name:mx.nd.zeros((75,)),
-                              cell.base_cell.r_cell.h2h_weight.var().name:mx.nd.zeros((75, 25)),
-                              cell.base_cell.r_cell.h2h_bias.var().name:mx.nd.zeros((75,))})
-    expected_outputs = np.ones((10, 50))+5
-    assert np.array_equal(outputs[0].asnumpy(), expected_outputs)
-    assert np.array_equal(outputs[1].asnumpy(), expected_outputs)
-
-
-def test_stack():
```

Comment:
Should this test be adapted to testing the front-end infer_shape functions instead?